apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx

```diff
@@ -178,27 +115,9 @@ export async function middleware(req: NextRequest) {
 </TabPanel>
 </Tabs>
 
-We can now use our server-side Supabase client to fetch data in Server Components.
+## Supabase Provider
 
-```jsx
-import 'server-only'
-
-import { createClient } from '../../utils/supabase-server'
-
-// do not cache this page
-export const revalidate = 0
-
-export default async function ServerComponent() {
-  const supabase = createClient()
-  const { data } = await supabase.from('posts').select('*')
-
-  return <pre>{JSON.stringify({ data }, null, 2)}</pre>
-}
-```
-
-### Client-side
-
-We still need a Supabase instance client-side for authentication and realtime subscriptions. It is important, when using Supabase client-side, to have a single instance of a client. We can share this singleton instance across our components using providers and React context.
+All Client Components need to share a single instance of the Supabase client. We can wrap our application in a `<SupabaseProvider />` and use React Context to create a global Supabase instance.
```

Comment:
```suggestion
All Client Components need to share a single instance of the Supabase client in order to prevent auth session conflicts. We can wrap our application in a `<SupabaseProvider />` and use React Context to create a global Supabase instance.
```